gym/utils/option_parser.py

```diff
@@ -0,0 +1,53 @@
+"""
+Utility functions used for classic control environments.
+"""
+
+from typing import Optional, Union
+import numpy as np
+
+
+def verify_number(x: Union[int, float, np.ndarray]) -> bool:
+    """Verify parameter is a single number."""
+    if type(x) == int or type(x) == float:
+        # A single value that is either an int or a float.
+        return True
+    if type(x) == np.ndarray:
+        if ((np.issubdtype(x.dtype, np.floating) or
+             np.issubdtype(x.dtype, np.integer)) and
+            len(x.shape) == 0):
+            # A numpy single value that is either an int or a float.
+            return True
+    return False
+
+
+def maybe_parse_reset_bounds(default_low: float,
+                             default_high: float,
+                             limit_low: float,
+                             limit_high: float,
+                             options: Optional[dict] = None) -> [float, float]:
```

Comment:
i've put it as the first argument, but i've kept it as optional since when it is `None` it means we should return the `default` values.
if we made it required, then we'd have to check that `options is not None` in the caller and then assign default values, which i think would result in a fair bit of code duplication.
what do you think?

Reply:
There might be a confusion between two notions of optional. I'm referring to giving it a default value of `None`, which makes it "optional", in the sense that you can skip the argument when calling the function. You seem to be talking about it being "Optional" (notice the capital letter), which in python is the same as saying `Union[x, None]`

I'm saying it should still be `Optional[dict]`, but not "optional", because it always has to get an argument. This argument might very well be `None`, which is consistent with the original type of `options` in the external API (it's also `Optional[dict]`)